contracts/Dispatcher.sol

```diff
@@ -541,6 +596,84 @@ contract Dispatcher is IbcDispatcher, Ownable {
         emit Acknowledgement(address(receiver), packet.src.channelId, packet.sequence);
     }
 
+    /**
+     * @notice Handle the incentivized acknowledgement of an IBC packet by the counterparty
+     * @dev Verifies the given proof and calls the `onAcknowledgementPacket` function on the given `receiver` contract,
+     *   ie. the IBC dApp.
+     *   Prerequisite: the original packet is committed and not ack'ed or timed out yet.
+     * @param receiver The dApp contract that should handle the app-level packet acknowledgement
+     * @param packet The original IbcPacket data sent by the dApp
+     * @param incentivizedAck The incentivized acknowledgement from counterparty chain, where the relayer is the payee address on behalf of the forward relayer that delivered the packet
+     * @param proof The membership proof to verify the packet acknowledgement committed on Polymer chain
+     */
+    function incentivizedAcknowledgement(
+        IbcReceiver receiver,
+        IbcPacket calldata packet,
+        IncentivizedAckPacket calldata incentivizedAck,
+        Proof calldata proof
+    ) external {
+        // verify `receiver` is the original packet sender
+        require(portIdAddressMatch(address(receiver), packet.src.portId), 'Receiver is not the original packet sender');
+        // prove ack packet is on Polymer chain
+        require(
+            verifier.verifyMembership(
+                latestConsensusState,
+                proof,
+                'ack/packet/path',
+                'expected ack receipt hash on Polymer chain'
+            ),
+            'Fail to prove ack'
+        );
+        // verify packet has been committed and not yet ack'ed or timed out
+        bool hasCommitment = sendPacketCommitment[address(receiver)][packet.src.channelId][packet.sequence];
+        require(hasCommitment, 'Packet commitment not found');
+
+        // enforce ack'ed packet sequences always increment by 1 for ordered channels
+        Channel memory channel = portChannelMap[address(receiver)][packet.src.channelId];
+        if (channel.ordering == ChannelOrder.ORDERED) {
+            require(
+                packet.sequence == nextSequenceAck[address(receiver)][packet.src.channelId],
+                'Unexpected packet sequence'
+            );
+            nextSequenceAck[address(receiver)][packet.src.channelId] = packet.sequence + 1;
+        }
+
+        // distribute fee to relayer
+        require(incentivizedAck.relayer.length == 20, 'Invalid relayer address');
```

Comment:
Is there an `isAddress()` function or similar?

Reply:
I haven't found any